### PR TITLE
feat(highlight): add dynamodb infra for note storage

### PR DESCRIPTION
**Pull request by kschelonka**
## Goal
Add localstack and AWS dynamodb. Tested with terraform plan success, and docker-compose success.

## References

JIRA ticket:
* [INFRA-291]

[INFRA-291]: https://getpocket.atlassian.net/browse/INFRA-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,24 @@
 version: '3.1'
 services:
+  localstack:
+    image: localstack/localstack
+    env_file:
+      - .docker/local.env
+    ports:
+      - '4566:4566'
```
Comment (bassrock) [inline docker-compose.yml:8]: Not for now, but a note at somepoint we should start changing our localstack ports so we can run multiple subgraphs.

File: .aws/src/dynamodb.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  /**
+   * Sets up the dynamodb table where the notes for highlights will live
+   * @private
+   */
+  private setupHighlightNotesTable() {
+    // note that this config is mirrored in .docker/localstack/dynamodb/
+    // if config changes here, that file should also be updated
+    return new ApplicationDynamoDBTable(this, `highlight-notes`, {
```
Comment (bassrock) [inline .aws/src/dynamodb.ts:24]: I continue to be amazed at tf modules payoffs.

**Review by bassrock — APPROVED** [review]
👍

**Comment by kschelonka** [comment]
Codebuild succeeded but the hook is not updating. I'm going to go ahead and merge.

**Pull request merged**